Build failure with GCC 10: multiple definition of `gt_cstr_nofree_ulp_hashtype'

Hi, I'm getting this build error with GCC 10:

```
/usr/bin/ld: obj/src/mgth/mg_combinedscore.o:./src/mgth/metagenomethreader.h:224: multiple definition of `gt_cstr_nofree_ulp_hashtype'; obj/src/mgth/metagenomethreader.o:./src/mgth/metagenomethreader.h:224: first defined here
/usr/bin/ld: obj/src/mgth/mg_compute_gene_prediction.o:./src/mgth/metagenomethreader.h:224: multiple definition of `gt_cstr_nofree_ulp_hashtype'; obj/src/mgth/metagenomethreader.o:./src/mgth/metagenomethreader.h:224: first defined here
/usr/bin/ld: obj/src/mgth/mg_computepath.o:./src/mgth/metagenomethreader.h:224: multiple definition of `gt_cstr_nofree_ulp_hashtype'; obj/src/mgth/metagenomethreader.o:./src/mgth/metagenomethreader.h:224: first defined here
/usr/bin/ld: obj/src/mgth/mg_outputwriter.o:./src/mgth/metagenomethreader.h:224: multiple definition of `gt_cstr_nofree_ulp_hashtype'; obj/src/mgth/metagenomethreader.o:./src/mgth/metagenomethreader.h:224: first defined here
/usr/bin/ld: obj/src/mgth/mg_reverse.o:./src/mgth/metagenomethreader.h:224: multiple definition of `gt_cstr_nofree_ulp_hashtype'; obj/src/mgth/metagenomethreader.o:./src/mgth/metagenomethreader.h:224: first defined here
/usr/bin/ld: obj/src/mgth/mg_xmlparser.o:./src/mgth/metagenomethreader.h:224: multiple definition of `gt_cstr_nofree_ulp_hashtype'; obj/src/mgth/metagenomethreader.o:./src/mgth/metagenomethreader.h:224: first defined here
```

Is the declaration of `gt_cstr_nofree` in `src/mgth/metagenomethreader.h` maybe missing an `extern` ?

--- a/src/ltr/ltrdigest_file_out_stream.c
+++ b/src/ltr/ltrdigest_file_out_stream.c
@@ -1,5 +1,5 @@
 /*
-  Copyright (c) 2008-2015 Sascha Steinbiss <sascha@steinbiss.name>
+  Copyright (c) 2008-2020 Sascha Steinbiss <sascha@steinbiss.name>
   Copyright (c) 2008-2013 Center for Bioinformatics, University of Hamburg
 
   Permission to use, copy, modify, and distribute this software for any
@@ -33,6 +33,7 @@
 #include "core/symbol.h"
 #include "core/undef_api.h"
 #include "core/unused_api.h"
+#include "core/warning_api.h"
 #include "extended/extract_feature_sequence.h"
 #include "extended/feature_node.h"
 #include "extended/feature_node_iterator_api.h"
@@ -416,6 +417,8 @@ int gt_ltrfileout_stream_next(GtNodeStream *ns, GtGenomeNode **gn, GtError *err)
 
     if (!had_err) {
       GtRange rng;
+      int ret = 0;
+
       ls->element.seqid = gt_calloc((size_t) ls->seqnamelen+1, sizeof (char));
       (void) snprintf(ls->element.seqid,
                       GT_MIN((size_t) gt_str_length(sdesc),
@@ -425,11 +428,15 @@ int gt_ltrfileout_stream_next(GtNodeStream *ns, GtGenomeNode **gn, GtError *err)
       if (gt_str_length(sdesc) > (GtUword) ls->seqnamelen)
         ls->element.seqid[ls->seqnamelen] = '\0';
 
-      (void) gt_ltrelement_format_description(&ls->element,
-                                              ls->seqnamelen,
-                                              desc,
-                                              (size_t) (GT_MAXFASTAHEADER-1));
+      ret = gt_ltrelement_format_description(&ls->element,
+                                             ls->seqnamelen,
+                                             desc,
+                                             (size_t) (GT_MAXFASTAHEADER-1));
+      if (ret < 0) {
+        gt_warning("FASTA header truncated: %s", desc);
+      }
       gt_str_delete(sdesc);
+
 
       /* output basic retrotransposon data */
       lltr_rng = gt_genome_node_get_range((GtGenomeNode*) ls->element.leftLTR);

--- a/src/mgth/metagenomethreader.h
+++ b/src/mgth/metagenomethreader.h
@@ -221,7 +221,7 @@ typedef struct
 } ParseStruct;
 
 /* specific access mode of queryhash */
-DECLARE_HASHMAP(char *, gt_cstr_nofree, GtUword *, ulp,,)
+DECLARE_HASHMAP(char *, gt_cstr_nofree, GtUword *, ulp, extern,)
 DECLARE_SAFE_DEREF(GtUword *, ulp)
 
 /* Funktion, mit der der Metagenomethreader gestartet wird

